ballerina/tests/data_binding_publish_tests.bal

```diff
@@ -617,6 +636,637 @@ service object {
+    groups: ["nats-basic"]
+}
+public function testDataBindingXmlPayloadPublish() {
+    xml messageToSend = xml `<start><Person><name>wso2</name><location>col-03</location></Person><Person><name>wso2</name><location>col-03</location></Person></start>`;
+    Client? newClient = clientObj;
+    if newClient is Client {
+        Listener? sub = listenerObj;
+        if sub is Listener {
+            checkpanic sub.attach(consumerServiceXmlPayloadPublish);
+            checkpanic sub.'start();
+            checkpanic newClient->publishMessage({ content: messageToSend, subject: DATA_BINDING_PUBLISH_SUBJECT });
+            int timeoutInSeconds = 120;
+            // Test fails in 2 minutes if it is failed to receive the message
+            while timeoutInSeconds > 0 {
+                if receivedXmlPayloadValuePublish !is "" {
+                    string receivedMessage = receivedXmlPayloadValuePublish;
+                    test:assertEquals(receivedMessage, messageToSend.toString(), msg = "Message received does not match.");
+                    break;
+                } else {
+                    runtime:sleep(1);
+                    timeoutInSeconds = timeoutInSeconds - 1;
+                }
+            }
+            checkpanic sub.detach(consumerServiceXmlPayloadPublish);
+            if timeoutInSeconds == 0 {
+                test:assertFail("Failed to receive the message for 2 minutes.");
+            }
+        } else {
+            test:assertFail("NATS Connection creation failed.");
+        }
+    } else {
+        test:assertFail("NATS Connection creation failed.");
+    }
+}
+
+Service consumerServiceXmlPayloadPublish =
+@ServiceConfig {
+    subject: DATA_BINDING_PUBLISH_SUBJECT
+}
+service object {
+    remote function onMessage(xml payload) {
+        string|error message = payload.toString();
+        if message is string {
+            receivedXmlPayloadValuePublish = message;
+            log:printInfo("Message Received: " + message);
+        }
+    }
+};
+
+@test:Config {
+    dependsOn: [testDataBindingXmlPayloadPublish],
+    groups: ["nats-basic"]
+}
+public function testDataBindingJsonPayloadPublish() {
+    json messageToSend = jsonData;
+    Client? newClient = clientObj;
+    if newClient is Client {
+        Listener? sub = listenerObj;
+        if sub is Listener {
+            checkpanic sub.attach(consumerServiceJsonPayloadPublish);
+            checkpanic sub.'start();
+            checkpanic newClient->publishMessage({ content: messageToSend, subject: DATA_BINDING_PUBLISH_SUBJECT });
+            int timeoutInSeconds = 120;
+            // Test fails in 2 minutes if it is failed to receive the message
+            while timeoutInSeconds > 0 {
+                if receivedJsonPayloadValuePublish !is () {
+                    json receivedMessage = receivedJsonPayloadValuePublish;
+                    test:assertEquals(receivedMessage, messageToSend, msg = "Message received does not match.");
```

Comment:
```suggestion
                    test:assertEquals(receivedJsonPayloadValuePublish, messageToSend, msg = "Message received does not match.");
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-nats/pull/427/commits/433577ffabea99c86a5ec34828b469bce521fc85